Doc/c-api/tuple.rst

```diff
@@ -35,6 +35,10 @@ Tuple Objects
 
    Return a new tuple object of size *len*, or *NULL* on failure.
 
+   .. versionchanged:: 3.7
+
+      ``PyTuple_New(0)`` always succeeds and never returns *NULL*.
```

Comment:
Do we need to make this a public API promise? IMHO it looks more like an implementation detail and may be inclined to change in future?

Reply:
The promise is essential to its usefulness.  Also, it is an easy thing for an implementation to guarantee.